src/tools/sfdk/sdkmanager.cpp

```diff
@@ -1348,15 +1348,37 @@ bool SdkManager::mapEnginePaths(QString *program, QStringList *arguments, QStrin
         {cleanSharedSrc, Constants::BUILD_ENGINE_SHARED_SRC_MOUNT_POINT, caseInsensitiveOnWindows}
 #endif
     };
+    *program = QDir::fromNativeSeparators(*program);
+
     for (const Mapping &mapping : mappings) {
         qCDebug(sfdk) << "Mapping" << mapping.hostPath << "as" << mapping.enginePath;
         program->replace(mapping.hostPath, mapping.enginePath, mapping.cs);
-        arguments->replaceInStrings(mapping.hostPath, mapping.enginePath, mapping.cs);
+        QRegularExpression::PatternOption reCaseSensitivity = (mapping.cs == Qt::CaseInsensitive)
+                ? QRegularExpression::CaseInsensitiveOption : QRegularExpression::NoPatternOption;
+        QRegularExpression re(QString("(?:%1|%2)([^:;,'\"]*)")
+                              .arg(QRegularExpression::escape(mapping.hostPath))
+                              .arg(QRegularExpression::escape(QDir::toNativeSeparators(mapping.hostPath))),
+                              reCaseSensitivity);
```

Comment:
I would move this into the lambda and move the lambda up just below the qCDebug to not disturb the higher level flow. I would consider adding some adjective to the name to express its special nature - maybe `mapGreedy`.

Reply:
done